src/lib.rs

```diff
@@ -96,15 +96,23 @@
 //! [3]: https://research.nccgroup.com/2020/02/26/public-report-rustcrypto-aes-gcm-and-chacha20poly1305-implementation-review/
 //! [4]: https://www.mobilecoin.com/
 
-#![no_std]
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
I think this is wrong, the crate is unconditionally no_std so why make it more complicated?

Reply:
We should really try to keep the same feature set as upstream to ease maintenance

Reply:
Upstream has a `std` feature, but no internal usage that actually requires `Vec`, that's why it gets away with being unconditionally `no_std`---this isn't adding any additional features to upstream.

Reply:
i see, i didn't realize upstream has an std feature